MantidPlot/src/ProjectRecovery.cpp

```diff
@@ -0,0 +1,354 @@
+#include "ProjectRecovery.h"
+
+#include "ApplicationWindow.h"
+#include "Folder.h"
+#include "ProjectSerialiser.h"
+
+#include "MantidAPI/FileProperty.h"
+#include "MantidAPI/FrameworkManager.h"
+#include "MantidKernel/ConfigService.h"
+#include "MantidKernel/Logger.h"
+#include "MantidKernel/UsageService.h"
+
+#include "boost/optional.hpp"
+
+#include "Poco/DirectoryIterator.h"
+#include "Poco/NObserver.h"
+
+#include "Poco/Path.h"
+
+#include <QMetaObject>
+
+#include <chrono>
+#include <condition_variable>
+#include <ctime>
+#include <iomanip>
+#include <iostream>
+#include <mutex>
+#include <string>
+#include <thread>
+
+namespace {
+// Config helper methods
+
+template <typename T>
+boost::optional<T> getConfigValue(const std::string &key) {
+  T returnedValue;
+
+  int valueIsGood =
+      Mantid::Kernel::ConfigService::Instance().getValue<T>(key, returnedValue);
+
+  if (valueIsGood != 1) {
+    return boost::optional<T>{};
+  }
+
+  return boost::optional<T>{returnedValue};
+}
+
+boost::optional<bool> getConfigBool(const std::string &key) {
+  auto returnedValue = getConfigValue<std::string>(key);
+  if (!returnedValue.is_initialized()) {
+    return boost::optional<bool>{};
+  }
+
+  return returnedValue->find("true") != std::string::npos;
+}
+
+/// Returns a string to the current top level recovery folder
+std::string getRecoveryFolder() {
+  static std::string recoverFolder =
+      Mantid::Kernel::ConfigService::Instance().getAppDataDir() + "/recovery/";
+  return recoverFolder;
+}
+
+/// Gets a formatted timestamp
+std::string getTimeStamp() {
+  const char *formatSpecifier = "%Y-%m-%d %H-%M-%S";
```

Comment:
Maybe this is a personal thing but spaces in these paths can cause confusion and be slightly harder to work with in the terminal. Could we use `T` as a separator for than a space as in the ISO 8601 standard?